File: 1480.running-sum-of-1-d-array.js
/*
 * @lc app=leetcode id=1480 lang=javascript
 *
 * [1480] Running Sum of 1d Array
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {

    let runningSumArr = [];

    runningSumArr.push(nums[0]);

    for(let i = 1; i < nums.length; i++){
        runningSumArr.push(runningSumArr[i-1] + nums[i]);
    }

    return runningSumArr;
    
};
// @lc code=end

File: 1480.running-sum-of-1-d-array.java
/*
 * @lc app=leetcode id=1480 lang=java
 *
 * [1480] Running Sum of 1d Array
 */

// @lc code=start
class Solution {
    public int[] runningSum(int[] nums) {
        
        int[] runningSumArr = new int[nums.length];

        runningSumArr[0] = nums[0];

        for(int i = 1; i < nums.length; i++){
            runningSumArr[i] = runningSumArr[i-1] + nums[i];
        }

        return runningSumArr;
    }
}
// @lc code=end

